src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,385 @@
+}
+
+interface InputProps {
+  focused: boolean
+}
+
+const Input = (
+  props: InputProps & JSX.IntrinsicElements['input']
+): React.ReactElement => {
+  const { focused, ...inputProps } = props
+  const inputRef = useRef<HTMLInputElement>(null)
+  useEffect(() => {
+    if (focused && inputRef.current) {
+      inputRef.current.focus()
+    }
+  })
+
+  return (
+    <input
+      aria-autocomplete="list"
+      aria-controls=""
+      type="text"
+      {...inputProps}
+      autoCapitalize="off"
+      autoComplete="off"
+      ref={inputRef}
+    />
+  )
+}
+
+export const ComboBox = (
+  props: ComboBoxProps //& JSX.IntrinsicElements['select']
+): React.ReactElement => {
+  const {
+    id,
+    name,
+    className,
+    options,
+    defaultValue,
+    setFieldValue,
+    assistiveHint,
+    ...selectProps
+  } = props
+
+  let defaultLabel = ''
+  if (defaultValue) {
+    const defaultOption = options.find((opt: ComboBoxOption): boolean => {
+      return opt.id === defaultValue
+    })
+    if (defaultOption) {
+      defaultLabel = defaultOption.label
+    }
+  }
+
+  interface State {
+    isOpen: boolean
+    selectedOption: ComboBoxOption | undefined
+    focusedOption: ComboBoxOption | undefined
+    focusMode: FocusMode
+    filter: string | undefined
+    filteredOptions: ComboBoxOption[]
+    inputValue: string
+  }
+
+  type Action =
+    | {
+        type: 'SELECT_OPTION'
+        option: ComboBoxOption
+      }
+    | {
+        type: 'CLEAR'
+      }
+    | {
+        type: 'OPEN_LIST'
+      }
+    | {
+        type: 'CLOSE_LIST'
+      }
+    | {
+        type: 'FOCUS_OPTION'
+        option: ComboBoxOption
+      }
+    | {
+        type: 'UPDATE_FILTER'
+        filter: string
+      }
+
+  function reducer(state: State, action: Action): State {
+    switch (action.type) {
+      case 'SELECT_OPTION':
+        props.setFieldValue(props.name, action.option.id)
+        return {
+          ...state,
+          isOpen: false,
+          selectedOption: action.option,
+          focusMode: FocusMode.Input,
+          inputValue: action.option.label,
+        }
+      case 'UPDATE_FILTER':
+        return {
+          ...state,
+          isOpen: true,
+          filter: action.filter,
+          filteredOptions: options.filter(optionFilter(action.filter)),
+        }
+      case 'OPEN_LIST':
+        return {
+          ...state,
+          isOpen: true,
+          focusMode: FocusMode.Input,
+          focusedOption: undefined,
+        }
+      case 'CLOSE_LIST':
+        return {
+          ...state,
+          isOpen: false,
+          focusMode: FocusMode.Input,
+          focusedOption: undefined,
+        }
+      case 'FOCUS_OPTION':
+        return {
+          ...state,
+          focusedOption: action.option,
+          focusMode: FocusMode.Item,
+        }
+      case 'CLEAR':
+        return {
+          ...state,
+          inputValue: '',
+          selectedOption: undefined,
```

Comment:
Also need to reset `filteredOptions` on this action to `props.action`

Reply:
I added a test stub to cover this case.